pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -794,7 +868,8 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 }
 
 func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.ControllerConfig, role, releaseImage string,
-	insecureRegs, registriesBlocked, policyBlocked, allowedRegs, searchRegs []string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) (*ign3types.Config, error) {
+	insecureRegs, registriesBlocked, policyBlocked, allowedRegs, searchRegs []string,
+	idmsRules []*apicfgv1.ImageDigestMirrorSet, itmsRules []*apicfgv1.ImageTagMirrorSet) (*ign3types.Config, error) {
```

Comment:
Non-blocking: This parameter list is getting very impractical.

Maybe there should be a `struct registriesConfig { insecure, registriesBlocked, policyBlocked, allowed, search, IDMS, ITMS, ICSP }` that a lot of the code could pass around as a single unit. Potentially factor out constructing this data out of the main line of `syncImageConfig` into a helper; and maybe make things like `getValidBlockedAndAllowedRegistries` a method (or a constructor?) on this struct.

That might also help respecting, rather than ignoring, the `// nolint: gocyclo` warning.

(Such a refactoring should almost certainly be a separate commit … or, well, maybe a separate PR entirely)

Reply:
I prefer refactor this in a separate pr.